IPs/RSKIP540.md

```diff
@@ -0,0 +1,129 @@
+
+**New behavior:**
+- The method builds a peg-out transaction using available UTXOs in the Bridge and current peg-out requests
+- It adds an extra peg-out request for the minimum peg-out value (as defined in RSKIP219 [[3]](#references)) instead of 1 BTC
+- Returns the estimated fees for this transaction
+
+### 2. New method `getEstimatedFeesForPegOutAmount(uint256 pegOutAmount)`
+
+A new method is added that allows users to estimate fees for a specific peg-out amount. This method builds a peg-out transaction similar to `getEstimatedFeesForNextPegOutEvent()` but uses the provided `pegOutAmount` parameter for the hypothetical peg-out request.
+
+**Method signature:**
+```
+function getEstimatedFeesForPegOutAmount(uint256 pegOutAmount) public view returns (uint256);
+```
+
+**Behavior:**
+- This method builds a peg-out transaction using available UTXOs in the Bridge and current peg-out requests
+- It adds an extra peg-out request using the provided `pegOutAmount` parameter as the amount
+- Returns the estimated fees for this transaction
+- The `pegOutAmount` parameter must be greater than or equal to the minimum peg-out value as defined in RSKIP219 [[3]](#references)
+
+**Validation:**
+- The `pegOutAmount` parameter must be greater than or equal to the minimum peg-out value
+- If the validation fails, the method should revert with an appropriate error message
+
+### Important Note on Fee Estimation Accuracy
+
+**The fees returned by both methods are estimates and are not guaranteed to match the actual fees charged when the peg-out event occurs.**
+
+Several conditions may change between the time the estimation is made and when the actual peg-out transaction is created, which can affect the final fee cost:
+
+- **New peg-out requests**: Other users may add peg-out requests to the queue, changing the number of outputs in the transaction
+- **Fee per KB changes**: The fee per KB value configured in RSK may be updated, affecting the transaction fee calculation
+- **UTXO availability**: The specific UTXOs available at estimation time may differ from those available when the peg-out event is executed
+
+Users and applications should be aware that the estimated fees are based on the current state of the Bridge at the time of the call and may differ from the actual fees charged. The estimates should be used as a guide rather than a guarantee.
+
+## Rationale
+
+### Using Minimum Peg-out Value
+
+Using the minimum peg-out value instead of 1 BTC for the hypothetical peg-out request in `getEstimatedFeesForNextPegOutEvent()` provides several benefits:
+- **More realistic estimates**: The minimum peg-out value better represents typical user transactions, especially for smaller amounts
+- **Conservative approach**: Using the minimum ensures the fee estimate works for all valid peg-out amounts
+- **Better UX**: Users performing small peg-outs will receive more accurate fee estimates
+
+### New Method `getEstimatedFeesForPegOutAmount()`
```

Comment:
Looks confusing to me, see the new method being referred to with `...()` empty, with no params. Let's mention with param or only the name `getEstimatedFeesForPegOutAmount`, what do you think?